fili-core/src/main/java/com/yahoo/bard/webservice/web/apirequest/DataApiRequestImpl.java

```diff
@@ -1252,14 +1252,24 @@ protected void validateApiHavings(
             if (sortDirectionMap == null) {
                 return metricSortColumns;
             }
+
+            Map<String, LogicalMetric> requestedMetrics = logicalMetrics.stream()
+                    .collect(Collectors.toMap(
+                            LogicalMetric::getName,
+                            x -> x
```

Comment:
Pretty sure there's a `java.util.Function.identity()` you can use here instead.

Reply:
I think there is some (very small) debate about which is preferable. Personally I prefer `x -> x`, though of course it doesn't make a big difference. The main selling point to me is `x -> x` is slightly better when debugging.

Reply:
https://stackoverflow.com/questions/27524445/does-a-lambda-expression-create-an-object-on-the-heap-every-time-its-executed/27524543#27524543

Reply:
Why is it better while debugging?  If we call into the Function.identity method we'll see the callstack going into it won't we?

I don't really care between these options.

Reply:
"step into" option shows where the function is sourced from, Function.identity will step into that static definition.

Reply:
I don't have a strong opinion. If x -> x makes your life easier, that's fine.